### ISSUE TITLE
Unable to install Google Cloud SDK anymore on "sudo" environment

### ISSUE THREAD

**Issue by florenthemmi** [travis-ci/travis-ci#9200]
Hello everyone,

Since this morning (**was working yesterday**), I have an error during Google Cloud SDK installation on "sudo" environment (`sudo: required`). I do not think it is related to Google, since this is working fine with `sudo: false`.

When debugging my build, I have run the following command to catch the error, here the output during installation:
```
...

╔════════════════════════════════════════════════════════════╗
╠═ Creating backup and activating new installation          ═╣
╚════════════════════════════════════════════════════════════╝

Performing post processing steps...
failed.
Traceback (most recent call last):
  File "/home/travis/google-cloud-sdk/bin/bootstrapping/install.py", line 214, in <module>
    main()
  File "/home/travis/google-cloud-sdk/bin/bootstrapping/install.py", line 192, in main
    Install(pargs.override_components, pargs.additional_components)
  File "/home/travis/google-cloud-sdk/bin/bootstrapping/install.py", line 134, in Install
    InstallOrUpdateComponents(to_install, update=update)
  File "/home/travis/google-cloud-sdk/bin/bootstrapping/install.py", line 177, in InstallOrUpdateComponents
    ['--quiet', 'components', verb, '--allow-no-backup'] + component_ids)
  File "/home/travis/google-cloud-sdk/lib/googlecloudsdk/calliope/cli.py", line 823, in Execute
    self._HandleAllErrors(exc, command_path_string, specified_arg_names)
  File "/home/travis/google-cloud-sdk/lib/googlecloudsdk/calliope/cli.py", line 861, in _HandleAllErrors
    exceptions.HandleError(exc, command_path_string, self.__known_error_handler)
  File "/home/travis/google-cloud-sdk/lib/googlecloudsdk/calliope/cli.py", line 797, in Execute
    resources = calliope_command.Run(cli=self, args=args)
  File "/home/travis/google-cloud-sdk/lib/googlecloudsdk/calliope/backend.py", line 757, in Run
    resources = command_instance.Run(args)
  File "/home/travis/google-cloud-sdk/lib/surface/components/update.py", line 106, in Run
    version=args.version)
  File "/home/travis/google-cloud-sdk/lib/googlecloudsdk/core/updater/update_manager.py", line 987, in Update
    self._PostProcess(snapshot=diff.latest)
  File "/home/travis/google-cloud-sdk/lib/googlecloudsdk/core/updater/update_manager.py", line 1473, in _PostProcess
    err_func=log.file_only_logger.debug)
  File "/home/travis/google-cloud-sdk/lib/googlecloudsdk/core/execution_utils.py", line 286, in Exec
    env = _GetToolEnv(env=env)
  File "/home/travis/google-cloud-sdk/lib/googlecloudsdk/core/execution_utils.py", line 145, in _GetToolEnv
    env, p.EnvironmentName(), p.Get(required=False, validate=False))
  File "/home/travis/google-cloud-sdk/lib/googlecloudsdk/core/properties.py", line 1557, in Get
    required)
  File "/home/travis/google-cloud-sdk/lib/googlecloudsdk/core/properties.py", line 1856, in _GetProperty
    value = _GetPropertyWithoutDefault(prop, properties_file)
  File "/home/travis/google-cloud-sdk/lib/googlecloudsdk/core/properties.py", line 1894, in _GetPropertyWithoutDefault
    value = callback()
  File "/home/travis/google-cloud-sdk/lib/googlecloudsdk/core/credentials/store.py", line 217, in GetAccount
    return c_gce.Metadata().DefaultAccount()
  File "/home/travis/google-cloud-sdk/lib/googlecloudsdk/core/credentials/gce.py", line 85, in DefaultAccount
    http_errors_to_ignore=(404,))
  File "/home/travis/google-cloud-sdk/lib/googlecloudsdk/core/util/retry.py", line 289, in DecoratedFunction
    six.reraise(to_reraise[0], to_reraise[1], to_reraise[2])
  File "/home/travis/google-cloud-sdk/lib/googlecloudsdk/core/util/retry.py", line 159, in TryFunc
    return func(*args, **kwargs), None
  File "/home/travis/google-cloud-sdk/lib/googlecloudsdk/core/credentials/gce.py", line 45, in _ReadNoProxyWithCleanFailures
    except urllib.request.HTTPError as e:
AttributeError: 'Module_six_moves_urllib_request' object has no attribute 'HTTPError'
```

Here my `.travis.yml` for reference:
```yml
dist: trusty
sudo: required

language: python
python:
  - "2.7"
cache:
  pip: true
  yarn: true
  directories:
    - front/node_modules

addons:
  chrome: stable

env:
  - PROJECT_ID=xxx

install:
  - nvm install 8
  - ./prepare-dev
  - echo "Installing google-cloud-sdk" && if [ ! -d "$HOME/google-cloud-sdk/bin" ]; then rm -rf "$HOME/google-cloud-sdk"; export CLOUDSDK_CORE_DISABLE_PROMPTS=1; curl https://sdk.cloud.google.com | bash 2>/dev/null >/dev/null; fi

before_script:
  - "export DISPLAY=:99.0"
  - "sh -e /etc/init.d/xvfb start"
  - source env/bin/activate
  - source "$HOME/google-cloud-sdk/path.bash.inc"

script:
  - ./bin/test_back.sh
  - ./bin/test_front.sh
```

Is someone having an idea?
Thanks in advance!

**Comment by BanzaiMan** [comment]
Do you have a build log URL that shows the problem you are describing here?

**Comment by florenthemmi** [comment]
My build are running on [travis-ci.com](https://travis-ci.com).
Please find attached the raw log.

[travis_job_109602948.log](https://github.com/travis-ci/travis-ci/files/1707260/travis_job_109602948.log)

**Comment by BanzaiMan** [comment]
We have not deployed any change over the time period in which you indicate the build's behavior changed.

Reading the logs, it looks like the error is coming purely from this command using the external command:

    echo "Installing google-cloud-sdk"; export CLOUDSDK_CORE_DISABLE_PROMPTS=1; curl https://sdk.cloud.google.com | bash >/dev/null

The files referenced in the stack trace are in `/home/travis/google-cloud-sdk`, indicating to me that the problem lies with Google's packaging. I just tried the command on my local machine, and it succeeded. Perhaps Google has fixed the packaging already. Could you try restarting it?

**Comment by florenthemmi** [comment]
Yes it is working locally.
I have restarted the build at least 5 times since this morning. 😉 

Please note that this working with `sudo: false`. Seems to be only failing with `sudo: required`...

**Comment by BanzaiMan** [comment]
This is most strange. We will take a further look. Thank you for the report.

**Comment by BanzaiMan** [comment]
I poked around a little bit, but I was unable to explain or resolve the issue. I opened the issue above ↑. Hopefully we can get his resolved soon.

**Comment by BanzaiMan** [comment]
Looks like this has been reported as https://issuetracker.google.com/issues/73044966.

**Comment by BanzaiMan** [comment]
https://issuetracker.google.com/issues/73044966#comment6 shows this workaround

    export CLOUDSDK_CORE_CHECK_GCE_METADATA=False

See https://travis-ci.org/BanzaiMan/travis_production_test/jobs/339076479

**Comment by florenthemmi** [comment]
Thanks for your help and the workaround. Indeed, it was a weird issue with the Google Cloud SDK...
I was thinking it was related to Travis CI because of the different behaviors between container and VM.

By the way, it has been resolved with Google Cloud SDK 188.0.1
I have tested this version and it is working as expected now.

So, I think we can close the issue.

**Comment by BanzaiMan** [comment]
Sweet!
